examples/models/llama2/export_llama_lib.py

```diff
@@ -510,11 +510,18 @@ def _export_llama(modelname, args) -> LLMEdgeManager:  # noqa: C901
         modelname = f"coreml_{modelname}"
 
     if args.qnn:
+        # pyre-ignore: Undefined import [21]: Could not find a module corresponding to import `executorch.backends.qualcomm.serialization.qnn_compile_spec_schema`
+        from executorch.backends.qualcomm.serialization.qnn_compile_spec_schema import (
+            QcomChipset,
+        )
         from executorch.extension.llm.custom_ops import model_sharding
 
         partitioners.append(
             get_qnn_partitioner(
-                args.use_kv_cache, args.pt2e_quantize, args.num_sharding
+                QcomChipset.SM8650,  # Llama 2 works only on SM8650
```

Comment:
It's still default to SM8650 right?

Reply:
I think we need to add a arg to export_llama script

Reply:
`get_qnn_partitioner` is only used here per https://github.com/search?q=repo%3Apytorch%2Fexecutorch%20get_qnn_partitioner&type=code.

I'm curious how other non-genai models are partitioned.

Reply:
As I said, this `partitioner_lib` is only used by llm and not used by anywhere. This is controlled by the compile spect when constructing the qnn partitioner. Should we just revert the change and use the default value?

Reply:
Are all llms going to always use SM8650 so you want to hard coded inside the partitioner_lib?

Reply:
@cccclai As titled this PR is to make it possible to get a qnn parititioner matching with the target device so that users can use it in their own lowering script for other LLMs. Sure today it's only used by the llama example in our codebase, and llama example wants it set to be SM8650 but I fail to understand why you prefer reverting the changes and hard-coded it back inside parititioner_lib after people are experiencing the chipset mismatch issue in https://github.com/pytorch/executorch/issues/4973

Reply:
I'll be fine with that. And notice that `extension/llm/export/partitioner_lib.py` is under `llm` folder, meaning it's only for llm. I don't think we will have bw to test on devices other than s24 before PTC.

Reply:
Also the real qnn partitioner is defined here https://github.com/pytorch/executorch/blob/b8a2cbd5a322aab7f58f34484db935d377692800/backends/qualcomm/partition/qnn_partitioner.py#L100-L114

The file you're changing is just a util function wrapped around qnn partitioner.